src/com.rs

```diff
@@ -15,7 +15,7 @@ impl<T> WeakPtr<T> {
         WeakPtr(ptr::null_mut())
     }
 
-    pub unsafe fn from_raw(raw: *mut T) -> Self {
+    pub fn from_raw(raw: *mut T) -> Self {
```

Comment:
This isn't sound. You can use this to turn a null pointer to an interface into a null reference to said interface -- this unsafe is the _only_ thing preventing corruption.

Reply:
I hadn't even looked for ref-coercing APIs, because I'd understood from the debug checks elsewhere and the `null` constructor that a non-null inner pointer was not an invariant violation. And yet, without it, you're right. This API was unsound, even before this PR. The `null` constructor right above this has the exact behavior you describe. 😅

Really, the only way to go from here and maintain is to either (1) remove `Deref` and provide a fallible alternative, or (2) rewrite the API to properly maintain a valid inner pointer as a variant.

After that, we can examine migrating to the `windows` ecosystem, which would obviate this entire noodle.

Thoughts?

Reply:
@cwfitzgerald: Did some minor revisions of my above comment. Not sure if you've read it yet, but I figured I'd let you know in case you'd draft a response already.

Reply:
Alright, going through the analysis, yeah ComPtr<T> is just unsound. wgpu uses the ability for these things to be null pretty often, so I am hesitant to ban nullability (this is in part due to the "make it null, then get a mut ptr to the pointer and set it" pattern that shows up often in d3d.

I'm going to snatch these changes in a larger "document the ComPtr PR"